core-bundle/contao/templates/analytics/analytics_google.html5

```diff
@@ -8,11 +8,12 @@ $GoogleAnalyticsId = 'XXXXX';
 if ('XXXXX' != $GoogleAnalyticsId && !$this->hasAuthenticatedBackendUser()): ?>
 
 <script async src="https://www.googletagmanager.com/gtag/js?id=<?= $GoogleAnalyticsId ?>"></script>
-<script>
+<script<?php if ($nonce = $this->nonce('script-src')): ?> nonce="<?= $nonce ?>"<?php endif; ?>>
```

Comment:
Can be optimized so it becomes less complex? Maybe just `<?php $this->addNonce('script-src') ?>` and if a nonce exists, it returns the `nonce="…"` tag?

Reply:
Personally I dislike having a function that returns a HTML attribute together with its value and leading space. Imho we should either use `HtmlAttributes` or the existing state of the PR.

Reply:
Me too, I think we should leave that as is and not introduce way to complex methods for the old templates. In Twig, we have more beautiful ways of achieving this and we're doing it there 👍

Reply:
I really dislike it, especially because it is added to a lot of templates. It really should be abstracted, so changing something does not require editing multiple templates.

Reply:
I have an idea, will create a PR tomorrow

Reply:
@leofeyer would you like https://github.com/fritzmg/contao/pull/5 better?

Reply:
Either way, we will always need a method that _only_ returns the nonce string and not a string containing the HTML attribute plus the nonce string, in case you want to use the nonce differently (e.g. with `HtmlAttributes`).

https://github.com/fritzmg/contao/pull/5 lgtm 👍 🙂

Reply:
> @leofeyer would you like [fritzmg#5](https://github.com/fritzmg/contao/pull/5) better?

Yes, I like that a lot! Although it seems you should replace all the `?>>>` with `?>>`, shouldn‘t you?

Reply:
Fixed.

Reply:
Merged :)